core/src/main/java/org/bitcoinj/wallet/DeterministicKeyChain.java

```diff
@@ -309,7 +309,7 @@ protected DeterministicKeyChain(DeterministicSeed seed) {
      * this method to watch an arbitrary fragment of some other tree, this limitation may be removed in future.
      */
     public DeterministicKeyChain(DeterministicKey watchingKey, long creationTimeSeconds) {
-        checkArgument(watchingKey.isPubKeyOnly(), "Private subtrees not currently supported");
```

Comment:
The issue is probably the way getPubOnly() works. It just reuses the non-pubOnly parent. Of course a pubOnly key that still has a private key in its parent isn't a pubOnly key… guess we need to rethink this more thoroughly.